Deletion of resource which doesn't exists breaks the pod

in [tektoncd/operator](https://github.com/tektoncd/operator), we have CRDs and CRs which we create during installation.
during uninstallation, we delete CRDs first and then rest of the resources. there is a time when we delete those CRs.
As the CRDs are already deleted it returns an error but the pod breaks here
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11e3438]

goroutine 414 [running]:
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetKind(...)
	/go/src/github.com/tektoncd/operator/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:226
github.com/manifestival/manifestival.okToDelete(0x0)
	/go/src/github.com/tektoncd/operator/vendor/github.com/manifestival/manifestival/manifestival.go:234 +0x38
github.com/manifestival/manifestival.Manifest.delete({{0xc000667000, 0x2b, 0x40}, {0x1e9c350, 0xc000418000}, {{0x1ebe080, 0xc0008cb890}, 0x0}}, 0x1e9c350, {0x0, ...})
	/go/src/github.com/tektoncd/operator/vendor/github.com/manifestival/manifestival/manifestival.go:179 +0xae
github.com/manifestival/manifestival.Manifest.Delete({{0xc000667000, 0x2b, 0x40}, {0x1e9c350, 0xc000418000}, {{0x1ebe080, 0xc0008cb890}, 0x0}}, {0x0, 0x0, ...})
```

because in deletion manifestival client here
https://github.com/tektoncd/operator/blob/main/vendor/github.com/manifestival/manifestival/manifestival.go#L175
doesn't return error and tries to check a nil object which breaks the pod


/kind bug

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 - No longer ignoring errors resulting from invalid YAML manifests [#88](https://github.com/manifestival/manifestival/issues/88)
 - No longer preventing cluster-scoped resources from being owned [#90](https://github.com/manifestival/manifestival/issues/90)
+- Fixed panic when unexpected errors occur during deletion [#92](https://github.com/manifestival/manifestival/issues/92)
 
 ### Added
 

--- a/manifestival.go
+++ b/manifestival.go
@@ -172,14 +172,15 @@ func (m Manifest) update(live, spec *unstructured.Unstructured, opts ...ApplyOpt
 // delete removes the specified object
 func (m Manifest) delete(spec *unstructured.Unstructured, opts ...DeleteOption) error {
 	current, err := m.get(spec)
-	if current == nil && err == nil {
+    if err != nil {
+        return err
+    }
+	if current == nil {
 		return nil
 	}
-
 	if !okToDelete(current) {
 		return nil
 	}
-
 	m.logResource("Deleting", spec)
 	return m.Client.Delete(spec, opts...)
 }

--- a/manifestival_test.go
+++ b/manifestival_test.go
@@ -287,6 +287,19 @@ func TestGenerateName(t *testing.T) {
 	assert(t, count, expected)
 }
 
+func TestDeleteError(t *testing.T) {
+	client := fake.Client{
+		fake.Stubs{
+			Get: func(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+				return nil, errors.NewBadRequest("TestDeleteError")
+			},
+		},
+	}
+	manifest, _ := NewManifest("testdata/k-s-v0.12.1.yaml", UseClient(client))
+	err := manifest.Delete()
+	assert(t, err.Error(), "TestDeleteError")
+}
+
 func assert(t *testing.T, actual, expected interface{}) {
 	t.Helper()
 	if actual == expected {

